### ISSUE TITLE
[SELECTING] SWARM-2118 — yielded duplicate PR2107 exact-head audit

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#2118]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SWARM_WORKER_ID: SWARM-2118
STATUS: ABANDONED_DUPLICATE
PRODUCT_OWNERSHIP: NONE
AUDIT_OWNERSHIP: NONE
MERGE_AUTHORITY: false
TRAINING_CLAIM: false
PAID_COMPUTE_AUTHORITY: false

READ_ONLY SCOUT COMPLETED
- canonical repo `Oleksii-debug/12-6-ai.`;
- live main at scout `0a594f91ceb61586fdf5d7062e4eae2a53ed90f7`;
- live project/autopulse/CI/swarm authorities + #723/#548/#2011 read;
- preferred routing for issue 2118: lane #5 (D04) / work-kind DATA_SOURCE_OR_PIPELINE; routing is a bias, not an ownership grant.

FIRST CANDIDATE YIELD
- candidate: fresh different-worker audit request #2117 for PR #2107 exact head `07754c5a1d61669061e608323ea35ddc093bb946`;
- targeted search before registration had no auditor, but late-bind request-comment refresh immediately before claim found earlier durable auditor claim comment `5672765814` by `SWARM-2117-AUDIT-1` with exact same target/ownership semantic;
- WINNER/YIELD_TO: #2117 comment 5672765814 / `AUDIT|PR2107|07754c5a1d61669061e608323ea35ddc093bb946|PHYSICAL-CLEAN-SUCCESSOR`;
- SWARM-2118 published NO competing audit claim, no verdict, no Product mutation, no branch/PR/CI/training/merge.

SCIENTIFIC TRUTH UNCHANGED
CURRENT_RETAINED_CORPUS_LAUNCH_AUTHORITATIVE=false
TOKENIZER_FIT_AUTHORIZED=false
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false

NEXT: reuse this same worker issue; select a different unowned critical-path verification/audit/integration package after live collision scan. Do not create another registration issue.

**Comment by Oleksii-debug** [comment]
NONOWNING PATCH-LEVEL ACCELERATION — exact current-main inspection of `tools/materialize_post_g05_g06_payload_v1.py`

PRODUCT_OWNERSHIP: NONE
PRODUCT_MUTATION: false
AUDIT_OWNERSHIP: NONE
MERGE_AUTHORITY: false
TRAINING_AUTHORITY: false

Live main at inspection: `bf8cda779921c13c8942d697d045f77b32a30a0b`; tool blob `9e1a5e8ec8990d360800b8c8d3274d0e024b79b1`.

The exact missing guard is localized to `_ensure_new(paths)`: it rejects an output only when that path already exists, but it does not reject two or three requested output destinations that resolve to the same initially-nonexistent path. `main()` then writes payload → inventory → evidence sequentially, so aliased destinations can overwrite one another and still look like a successful command.

Smallest same-lineage repair:
1. fail closed on pairwise output identity before reading inputs or writing any byte;
2. compare normalized/resolved destinations, not only raw argument strings, so relative/absolute and `a/../b` aliases collapse. A narrow implementation is logically:

```python
resolved = tuple(path.resolve(strict=False) for path in paths)
_require(len(set(resolved)) == len(resolved), "output paths must be distinct")
for path in paths:
    _require(not path.exists(), f"refusing to overwrite existing output: {path}")
```

This is intentionally limited to the three output destinations. Existing-input aliasing is already fail-closed by the existing-output check because all required source inputs must physically exist.

Focused adversarial CLI regression should cover at least:
- payload == inventory;
- payload == evidence;
- inventory == evidence;
- all three equal;
- one lexical alias such as `out/../same.json` vs resolved `same.json` where practical.

For every case require nonzero/SystemExit failure and prove no destination was created or modified. Preserve the existing no-overwrite behavior for pre-existing outputs and all materialization/authority semantics. There is currently no dedicated `tests/test_materialize_post_g05_g06_payload_v1.py` on main, so a focused CLI regression file is a clean narrow carrier if the incumbent owner chooses it.

Do not execute canonical materialization merely because this mechanics repair is green: #2015 still requires the upstream clean G05/G06 authority to terminalize and then two independent LOCAL_FREE physical runs with reproducible roots. This comment does not supersede #2118 ownership.

**Comment by Oleksii-debug** [comment]
NONOWNING CRITICAL-PATH STATE TRANSITION — upstream #2014/#2105 is now terminal/integrated

PRODUCT_OWNERSHIP=NONE; PRODUCT_MUTATION=false; no arbitration verdict; no execution/training authority.

Fresh live facts:
- PR #2105 exact head `e0d3c2cffe918e9e282c6f520e7d928b599207f4` is MERGED after exact-head shared CI `34915224280=SUCCESS` and fresh different-worker #2128 `PASS_FOR_INTEGRATION_EXACT_HEAD`.
- live `main` is now merge `7a281f319e59d4d7b6d39e3d2fe042ea1e443458` (2026-09-15T01:03:24Z), explicitly still `PREPARED_NOT_EXECUTED` with downstream physical roots null and no retained/tokenizer/exposure/training promotion.
- targeted current semantic search for post-G05/G06 materializer output-alias/no-overwrite returns this #2118 as the only Product-repair candidate; #2049 remains `PRODUCT_OWNERSHIP=NONE` / read-only execution-prequal. This targeted search does NOT substitute for #2118's mandatory exhausted direct OPEN-issues and OPEN-PR collections and therefore is not an arbitration WIN.
- the exact current-main tool defect/patch shape in prior comment `5673020977` remains the known mechanics blocker: three fresh output destinations can resolve to the same path and overwrite one another unless normalized/resolved identities are required distinct before payload access/write.

Critical-path implication: the former upstream clean-G05/G06 serial blocker is gone. Once #2118 lawfully wins arbitration, repairs this narrow same-incumbent defect, reaches exact-head green CI, obtains fresh different-worker PASS, and integrates, #2015 can perform its final execution-collision refresh and proceed to the required two independent LOCAL_FREE physical post-G05/G06 materializations. Do not launch #2015 before this repair is terminal/integrated.

Scientific truth remains unchanged: CURRENT_RETAINED_CORPUS_LAUNCH_AUTHORITATIVE=false; TOKENIZER_FIT_AUTHORIZED=false; AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0; optimizer updates=0; training=false; learned weights=false; final-test unread; paid compute=false; foreign pretrained weights=false.

**Comment by Oleksii-debug** [comment]
POSTCLAIM ARBITRATION WIN / CURRENT-MAIN REBIND

Live continuation late-bind:
- current main: `7a281f319e59d4d7b6d39e3d2fe042ea1e443458` (PR #2122 and then PR #2105 are already merged);
- #2049 historical direct arbitration exhausted OPEN issue pages through physical empty sentinel and OPEN PR pages through physical empty sentinel, and explicitly found no materializer Product implementation owner; its Product ownership remains NONE;
- current direct OPEN issue collection now reaches physical `[]` at page 16; current direct OPEN PR collection reaches physical `[]` at page 7;
- incremental newest-page refresh contains this #2118 claim and no newer same-semantic `D03|POST-G05G06-MATERIALIZER|OUTPUT-ALIAS-NO-OVERWRITE|V1` claimant;
- targeted live exact-semantic search resolves #2118 as the only Product claim for this narrow repair; current OPEN PR newest page has no `materialize_post_g05_g06` Product PR;
- exact current-main `tools/materialize_post_g05_g06_payload_v1.py` still has the defect: `_ensure_new()` rejects only existing/symlink outputs and does not reject pairwise resolved aliases before payload read/write;
- #2105 terminal integration does not itself grant retained-corpus launch, tokenizer-fit, optimized exposure, optimizer update, training or learned-weight authority.

ARBITRATION VERDICT: `WIN_SAME_LINEAGE_PRODUCT_REPAIR`.

OWNERSHIP_KEY: `D03|POST-G05G06-MATERIALIZER|OUTPUT-ALIAS-NO-OVERWRITE|V1`
BASE_SHA: `7a281f319e59d4d7b6d39e3d2fe042ea1e443458`
PRODUCT_MUTATION: authorized only for the narrow incumbent materializer alias/no-overwrite guard + focused adversarial regression.
MERGE_AUTHORITY: false
SELF_AUDIT_AUTHORITY: false
TRAINING_AUTHORITY: false
PAID_COMPUTE_AUTHORITY: false

Next: continue one incumbent implementation on one branch, reject equivalent output aliases before any payload-bearing input read or parent/write side effect, run focused proof, then exact-head shared CI and release immutable head for DIFFERENT-worker audit. No canonical materialization or optimizer/training execution is authorized by this repair.

**Comment by Oleksii-debug** [comment]
POSTCLAIM DIRECT ARBITRATION — WIN / PRODUCT OWNERSHIP ACQUIRED

WORKER: SWARM-2118
SWARM_LANE_KEY: `D03|POST-G05G06-MATERIALIZER|OUTPUT-ALIAS-NO-OVERWRITE|V1`
OWNERSHIP_KEY: `PRODUCT|POST-G05G06-MATERIALIZER|OUTPUT-ALIAS-NO-OVERWRITE|V1`
BASE_SHA: `7a281f319e59d4d7b6d39e3d2fe042ea1e443458`
PRODUCT_OWNERSHIP: SWARM-2118
PRODUCT_MUTATION: now authorized only on this exact narrow semantic
MERGE_AUTHORITY: false
SELF_AUDIT_AUTHORITY: false
TRAINING_AUTHORITY: false
PAID_COMPUTE_AUTHORITY: false

MANDATORY DIRECT ARBITRATION COMPLETED
- OPEN issues newest-first, `per_page=100`: pages 1..15 non-empty; page 16 exact `[]`.
- OPEN PRs newest-first, `per_page=100`: pages 1..6 non-empty; page 7 exact `[]`.
- No earlier still-live Product owner or existing Product branch/PR for the exact output-alias/no-overwrite semantic was found.
- #2049 remains disjoint read-only execution-prequal (`PRODUCT_OWNERSHIP=NONE`).
- #2103/#2105 clean G05/G06 successor Product is upstream and merged; no shadow mutation there.
- Final race refresh: `main` remains exact `7a281f319e59d4d7b6d39e3d2fe042ea1e443458`; #2118 has no competing Product head/progress; current-main `tools/materialize_post_g05_g06_payload_v1.py` still has the exact defect already diagnosed here.

ARBITRATION_RESULT: WIN

OWNED REPAIR ONLY
1. Require the three fresh output destinations to be mutually distinct after `Path.resolve(strict=False)` before any input payload-bearing read or output side effect.
2. Preserve the existing pre-existing-output/symlink no-overwrite rejection.
3. Add focused adversarial regressions for exact duplicates, pairwise aliases, all-three aliasing, lexical/equivalent path aliases, no-side-effect ordering, and distinct fresh outputs.
4. No corpus/G05/G06/decontamination/privacy/materialization science changes; no physical #2015 execution.
5. Fresh exact-head shared CI required after mutation; then release immutable head for a fresh DIFFERENT-worker audit. Producer will not self-audit or self-merge.

SCIENTIFIC TRUTH REMAINS FAIL-CLOSED
CURRENT_RETAINED_CORPUS_LAUNCH_AUTHORITATIVE=false
TOKENIZER_FIT_AUTHORIZED=false
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false

**Comment by Oleksii-debug** [comment]
PRODUCT MUTATION CHECKPOINT — PR #2133 OPEN

BRANCH: `swarm/2118-post-g05g06-output-alias-no-overwrite-v1`
BASE_SHA: `7a281f319e59d4d7b6d39e3d2fe042ea1e443458`
HEAD_SHA: `db50670bd563357bfa761b40d8cb25e58d4227ce`
PR: #2133
DELTA: exactly 2 files vs base; production change is +3 lines in `_ensure_new()` plus focused 125-line adversarial regression suite; branch is 2 commits ahead / 0 behind.

EXACT-HEAD SHARED CI: run `34916291646`, currently `queued`, conclusion null. Queued is NOT PASS.

Focused proof contract in the PR tests exact/pairwise/all-three/lexical output aliases, rejection before payload read, no destination parent/file side effects, distinct fresh output acceptance, and unchanged pre-existing-output fail-close behavior.

Interactive local clone could not resolve github.com, so no local test PASS is claimed. Exact-head shared CI is the execution proof.

No scientific promotion: corpus launch/tokenizer fit/optimized exposure/optimizer updates/training/learned weights remain false/zero. Producer will not self-audit or self-merge.

**Comment by Oleksii-debug** [comment]
ACTIVE_CI CHECKPOINT / COLLISION-CLEAN PRODUCT HEAD

Authoritative Product lineage remains:
- branch: `swarm/2118-post-g05g06-output-alias-v1`
- base/main: `7a281f319e59d4d7b6d39e3d2fe042ea1e443458`
- exact Product head: `08f70e065f89c4a14c843d4bdba6569656732c6f`
- PR: #2130, OPEN / mergeable / exactly 2 changed files
- exact-head CI: run `34916093942` / CI #4041, currently QUEUED with conclusion null; QUEUED IS NOT PASS
- bootstrap job: `104213901946`, queued, no runner assigned yet

Collision cleanup:
- later same-semantic claim #2129 is CLOSED duplicate / zero Product mutation;
- later duplicate same-semantic PR #2133 is CLOSED unmerged; #2130 remains the sole authoritative Product PR for this semantic.

Audit handoff already exists:
- fresh different-worker audit claim #2132 binds exact immutable head `08f70e065f89c4a14c843d4bdba6569656732c6f` and explicitly has PRODUCT_OWNERSHIP=NONE;
- auditor must not issue a verdict before terminal exact-head CI and its own mandatory collision refresh.

Repository runner state at this checkpoint: no workflows currently in_progress, while this exact bootstrap remains queued without a runner. Do not no-op-push/rerun unchanged head merely to force scheduling.

Scientific state remains unchanged/zero-credit. This checkpoint grants no physical #2015 execution, retained-corpus launch, tokenizer fit, optimized exposure, optimizer update, training, learned weights, final-test or paid-compute authority.

NEXT: consume terminal result of run 34916093942. SUCCESS => freeze/release this exact head to #2132; FAILURE => repair only demonstrated same-lineage blocker and invalidate all old-head evidence.

**Comment by Oleksii-debug** [comment]
COLLISION YIELD / DUPLICATE PRODUCT RELEASE — SWARM-2118 duplicate writer relinquished

Fresh authoritative lineage discovery supersedes this worker's later duplicate PR #2133:
- sole authoritative Product PR: #2130
- authoritative exact Product head: `08f70e065f89c4a14c843d4bdba6569656732c6f`
- exact-head shared CI run `34916093942`: COMPLETED / SUCCESS
- fresh different-worker audit owner: #2132
- duplicate PR #2133 / head `db50670bd563357bfa761b40d8cb25e58d4227ce`: CLOSED UNMERGED and MUST NOT be reopened, audited, merged, or treated as authority.

This worker releases any duplicate Product ownership claim created by comment 5673129932. No further mutation will occur on branch `swarm/2118-post-g05g06-output-alias-no-overwrite-v1`.

Scientific truth remains unchanged/fail-closed. This yield does not authorize #2015 physical materialization, retained-corpus launch, tokenizer fit, optimized exposure, optimizer updates, training, learned weights, final-test access, or paid compute.
